src/utils/svg.js

```diff
@@ -0,0 +1,199 @@
+  strokeColor,
+  width = 2,
+  height = 0,
+  drawStrokes = true,
+  drawFillShapes = true,
+  strokesWireframe = false,
+  fillShapesWireframe = false,
+}) => {
+  const svgData = await svgLoader.loadAsync(url)
+  debug().log('svg#getSVGGroup: svgData: ', svgData)
+  const paths = svgData.paths
+  const group = new THREE.Group()
+  const svgGroup = new THREE.Group()
+
+  for (let i = 0; i < paths.length; i++) {
+    const path = paths[i]
+    if (!fillColor) {
+      fillColor = path.userData.style.fill
+    }
+    if (!strokeColor) {
+      strokeColor = path.userData.style.stroke
+    }
+
+    if (drawFillShapes && fillColor !== undefined && fillColor !== 'none') {
+      const material = new THREE.MeshBasicMaterial({
+        color: new THREE.Color().setStyle(fillColor).convertSRGBToLinear(),
+        opacity: path.userData.style.fillOpacity,
+        transparent: true,
+        side: THREE.DoubleSide,
+        // depthWrite: false,
+        wireframe: fillShapesWireframe,
+      })
+      const shapes = SVGLoader.createShapes(path)
+
+      for (let j = 0; j < shapes.length; j++) {
+        const shape = shapes[j]
+        const geometry = new THREE.ShapeGeometry(shape)
+        const mesh = new THREE.Mesh(geometry, material)
+        group.add(mesh)
+      }
+    }
+
+    if (drawStrokes && strokeColor !== undefined && strokeColor !== 'none') {
+      const material = new THREE.MeshBasicMaterial({
+        color: new THREE.Color().setStyle(strokeColor).convertSRGBToLinear(),
+        opacity: path.userData.style.strokeOpacity,
+        transparent: true,
+        side: THREE.DoubleSide,
+        // depthWrite: false,
+        wireframe: strokesWireframe,
+      })
+
+      for (let j = 0, jl = path.subPaths.length; j < jl; j++) {
+        const subPath = path.subPaths[j]
+        const geometry = SVGLoader.pointsToStroke(subPath.getPoints(), path.userData.style)
+
+        if (geometry) {
+          const mesh = new THREE.Mesh(geometry, material)
+          group.add(mesh)
+        }
+      }
+    }
+  }
+
+  const groupBox3 = new THREE.Box3()
+  groupBox3.setFromObject(group)
+  debug().log('svg#getSVGGroup: groupBox3: ', groupBox3)
+  const groupSize = groupBox3.max.sub(groupBox3.min)
+  debug().log('svg#getSVGGroup: groupSize: ', groupSize)
+  let scaleX = 0
+  let scaleY = 0
+  if (width) {
+    scaleX = width / groupSize.x
+  }
+  if (height) {
+    scaleY = height / groupSize.y
+  }
+  if (!width) {
+    scaleX = scaleY
+  }
+  if (!height) {
+    scaleY = scaleX
+  }
+  if (!width && !height) {
+    scaleX = scaleY = 1
+  }
+  if (!width) {
+    width = scaleX * groupSize.x
+  }
+  if (!height) {
+    height = scaleY * groupSize.y
+  }
+  debug().log('svg#getSVGGroup: group scales: ', scaleX, scaleY, width, height)
+  group.scale.x = scaleX
+  group.scale.y = scaleY
+  group.position.x = -width
+  group.position.y = height
+  group.scale.y *= - 1
+  svgGroup.add(group)
+  return svgGroup
+}
+
+
+export const getSVGMesh = async ({
```

Comment:
Please use function with jsdoc for top-level funcs

Reply:
done